Include/Rover/Pointer.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef ROVER_POINTER_HPP
+#define ROVER_POINTER_HPP
+#include <type_traits>
+
+namespace Rover {
+
+  /** Trait to check whether a type is a pointer to a C++ object or POD
+      type. */
+  template<typename, typename = void>
+  struct is_object_pointer : std::false_type {};
+
+  template<typename T>
+  struct is_object_pointer<T, std::enable_if_t<std::is_pointer_v<T>>> :
+      std::true_type {};
```

Comment:
Fix indentation.

Reply:
It seems like I'd gotten a bit confused about indentation rules. The wiki article clearly states that, in case of line continuation, indentation is 2 spaces except if there is a clash with the following code (block opening), however, while browsing through Beam/Nexus code, I did not come across a single example where constructor initializer lists are indented with 2 spaces only, even when constructor body is empty. For example,
A::A(int x)
    : m_x(x) {} // always indented with 4 spaces
Is there a special rule for initializer lists so we don't treat them as line continuations?

Reply:
Yep good point, that's an error in Beam/Nexus. There is no special rule for initializer lists, it's just a bad habit.